### PR TITLE
use ceph storage class for pvc for postgresql

### PR DESCRIPTION
**Pull request by harshad16**
use ceph storage class for PVC for Postgresql.
Signed-off-by: Harshad Reddy Nalla <hnalla@redhat.com>

### CODE REVIEW

File: tasks/main.yml [hunk 1]
```diff
@@ -4,16 +4,21 @@
...
-      -p POSTGRESQL_PASSWORD="{{ postgresql_password }}" \
-      -p POSTGRESQL_DATABASE="{{ postgresql_database }}" \
-      -p VOLUME_CAPACITY="{{ postgresql_volume_capacity }}" \
-      -p POSTGRESQL_VERSION="{{ postgresql_version }}" | oc apply -f - --namespace "{{ namespace }}"
+  - name: "provision PostgreSQL"
+    include_tasks: postgresql.yml
+    vars:
+      STORAGE_CLASS_NAME: "{{ storage_class_name }}"
+      CEPH_PV_SECRET: "{{ ceph_pv_secret }}"
+      CEPH_PV_SECRET: "{{ ceph_pv_secret_name }}"
```
Comment (fridex) [inline tasks/main.yml]: `CEPH_PV_SECRET_NAME`

File: tasks/postgresql.yml [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+        database-password: "{{ POSTGRESQL_PASSWORD }}"
+        database-user: "{{ POSTGRESQL_USER }}"
+
+- name: "create service for postgresql"
+  k8s:
+    state: present
+    definition:
+      kind: Service
+      apiVersion: v1
+      metadata:
```
Comment (fridex) [inline tasks/postgresql.yml:29]: Label `app=thoth`.

File: tasks/postgresql.yml [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+        storageClassName: "{{ STORAGE_CLASS_NAME }}"
+
+- name: "create secret for ceph based persistent volume"
+  k8s:
+    state: present
+    definition:
+      kind: Secret
+      apiVersion: v1
+      type: kubernetes.io/rbd
+      metadata:
```
Comment (fridex) [inline tasks/postgresql.yml:75]: Label `app=thoth`.

File: tasks/postgresql.yml [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+      status:
+        loadBalancer: {}
+
+- name: "create presistant volume claim for postgresql"
+  k8s:
+    state: present
+    definition:
+      kind: PersistentVolumeClaim
+      apiVersion: v1
+      metadata:
```
Comment (fridex) [inline tasks/postgresql.yml:57]: I think we are fine if we do not provide any label here (so `oc delete --all -l app=thoth` keeps it). Ack?

File: tasks/postgresql.yml [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+      data:
+        key: "{{ CEPH_PV_SECRET }}"
+
+- name: "create deployment for postgresql"
+  k8s:
+    state: present
+    definition:
+      kind: DeploymentConfig
+      apiVersion: apps.openshift.io/v1
+      metadata:
```
Comment (fridex) [inline tasks/postgresql.yml:87]: Label `app=thoth`.

File: tasks/postgresql.yml (hunk 2 above)
Comment (fridex) [reply tasks/postgresql.yml:29]: Maybe worth to also add `component=postgres`, here and in other places.

File: tasks/postgresql.yml (hunk 4 above)
Comment (goern) [reply tasks/postgresql.yml:57]: +1

File: vars/main.yml [hunk 6]
```diff
@@ -8,3 +8,7 @@ postgresql_user: thoth
 postgresql_database: postgres
 postgresql_volume_capacity: "1Gi"
 postgresql_version: "9.6"
+storage_class_name: ceph-dyn-thoth-paas-prod-psi
+ceph_pv_secret_name: ceph-thoth-paas-prod-psi-admin-secret
+# placeholder
+ceph_pv_secret: ceph-dyn-thoth-paas-prod-psi
```
Comment (goern) [inline vars/main.yml]: should we remove this default to not accidentally configure some insecure value and force the user to provide a value?

File: tasks/main.yml (hunk 1 above)
Comment (harshad16) [reply tasks/main.yml]: Thanks :smile:

File: vars/main.yml (hunk 6 above)
Comment (harshad16) [reply vars/main.yml]: yes, that make sense :+1:

File: tasks/postgresql.yml (hunk 3 above)
Comment (harshad16) [reply tasks/postgresql.yml:75]: if we are gonna keep the pvc by not adding the label to the pvc, then we shouldn't add the label to pvc secret as well. WDYT?

Comment (fridex) [reply tasks/postgresql.yml:75]: Makes sense :+1:

**Review by goern — APPROVED** [review]

**Pull request merged**